### PR TITLE
drt: parallelize the standalone drc check

### PR DESCRIPTION
**Pull request by osamahammad21**
Signed-off-by: osamahammad21 <osama21@aucegypt.edu>

### CODE REVIEW

File: src/drt/src/TritonRoute.cpp [hunk 1]
```diff
@@ -809,16 +809,77 @@ void TritonRoute::checkDRC(const char* filename, int x1, int y1, int x2, int y2)
...
   frList<std::unique_ptr<frMarker>> markers;
-  for (auto& marker : gcWorker->getMarkers())
-    markers.push_back(std::make_unique<frMarker>(*marker));
+  std::map<std::tuple<Rect,
+                      frLayerNum,
+                      frConstraint*,
+                      frBlockObject*,
+                      frBlockObject*>,
+           frMarker*>
+      mapMarkers_;
```
Comment (maliberty) [inline src/drt/src/TritonRoute.cpp:849]: Local variables should not have an _ suffix.

File: src/drt/src/TritonRoute.cpp [hunk 2]
```diff
@@ -809,16 +809,77 @@ void TritonRoute::checkDRC(const char* filename, int x1, int y1, int x2, int y2)
...
+    }
+  }
   frList<std::unique_ptr<frMarker>> markers;
-  for (auto& marker : gcWorker->getMarkers())
-    markers.push_back(std::make_unique<frMarker>(*marker));
+  std::map<std::tuple<Rect,
+                      frLayerNum,
+                      frConstraint*,
+                      frBlockObject*,
+                      frBlockObject*>,
```
Comment (maliberty) [inline src/drt/src/TritonRoute.cpp:847]: A struct would be better than a tuple.  I can't guess what  this holds and names would make it clearer.

File: src/drt/src/TritonRoute.cpp [hunk 3]
```diff
@@ -809,16 +809,77 @@ void TritonRoute::checkDRC(const char* filename, int x1, int y1, int x2, int y2)
...
+      mapMarkers_;
+  omp_set_num_threads(MAX_THREADS);
+#pragma omp parallel for schedule(dynamic)
+  for (int i = 0; i < workers.size(); i++) {
+    workers[i]->init(design_.get());
+    workers[i]->main();
+#pragma omp critical
+    {
+      for (auto& marker : workers[i]->getMarkers()) {
+        Rect bbox = marker->getBBox();
```
Comment (maliberty) [inline src/drt/src/TritonRoute.cpp:858]: The markers might be inside a worker but outside the area being checked ('box').  Do you want to filter them?

Surely there is a better name than box

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/drt/src/TritonRoute.cpp (hunk 3 above)
Comment (osamahammad21) [reply src/drt/src/TritonRoute.cpp:858]: They are filtered out by reportDRC

**Review by maliberty — APPROVED** [review]

**Pull request merged**